src/main/scala/gestalt/Trees.scala

```diff
@@ -163,11 +162,19 @@ trait Trees extends Params with TypeParams with
   val TypeIdent: TypeIdentImpl
   trait TypeIdentImpl {
     def apply(name: String): TypeTree
+    def unapply(tpe: TypeTree): Option[String]
   }
 
   val TypeSelect: TypeSelectImpl
   trait TypeSelectImpl {
     def apply(qual: Tree, name: String): TypeTree
+    def unapply(tpe: TypeTree): Option[(Tree, String)]
+  }
+
+  val PathType: PathTypeImpl
+  trait PathTypeImpl {
+    def apply(qual: Option[Tree], name: String, targs: Seq[TypeTree]): TypeTree
+    def unapply(tpe: TypeTree) : Option[(Option[Tree], String, Seq[TypeTree])]
```

Comment:
The idea behind this was: `DottyToolbox.PathType.unapply` must contain full and correct implementations of `DottyToolbox.TypeIdent.unapply`, `DottyToolbox.TypeSelect.unapply` and `DottyToolbox.TypeApply.unapply`, might as well expose them in the API.

On second thought: other than this corner case extractors for `TypeTree`s are redundant. For information about type should look at `type`s not `TypeTree`s. I will make the change as suggested.